srv/src/main/java/my/bookshop/handlers/AdminServiceHandler.java

```diff
@@ -248,6 +260,43 @@ public void addBookToOrder(AddToOrderContext context) {
 		context.setResult(updatedOrder);
 	}
 
+	@On(entity = Csv_.CDS_NAME)
+	public void getCsvSingleton(CdsReadEventContext context) {
+		Csv csv = Csv.create();
+		csv.setDataType("text/csv");
+		Result result = ResultBuilder.selectedRows(Arrays.asList(csv)).result();
```

Comment:
you can also directly return the Result object, with this you don't need to use `context.getResult()`. If you remove the context from the arguments you need to add the READ event to the `@On` annotation though :)

Reply:
The `Result` object? Or can I also directly return the `Csv` object? I just tried

```java
@On(entity = Csv_.CDS_NAME, event = CdsService.EVENT_READ)
public Csv getCsvSingleton() {
    return Csv.create();
}
```
and that seems to work, too.

Reply:
Both is fine :) That way its even more elegant ;)